pyActigraphy/io/cwa/cwa_class.py

```diff
@@ -1,237 +1,513 @@
+import os
 from datetime import datetime
-from struct import Struct
+from datetime import timedelta
+from tools import DataReader
+
+import pandas
+from numpy.linalg import norm
+
+class CWA(object):
+    __slots__ = ["__endianess", "__stream", "__meta",
+                 "__blocks",
+                 "__device",
+                 "__name", "__uuid", "__startTime",
+                 "__duration", "__period",
+                 "__Accel", "__Gyro", "__Mag",
+                 "__Ascale", "__Gscale", "__Mscale",
+                 "__Others"]
+    __hardware = {0x00: "AX3",
+                  0xff: "AX3",
+                  0x17: "AX3",
+                  0x64: "AX6"}
+    __axes = {1: "Axyz",
+              2: "Gxyz/Axyz",
+              3: "Gxyz/Axyz/Mxyz"
+              }
+
+    @classmethod
+    def IsValidFile(cls, filename):
+        """
+        Checks if given file is a valid cwa file.
+        Checks is performed by name (do it have .cwa)
+        and by leading bits, which must be 'MD'
+
+        Parameters
+        ----------
+        filename: str
+            path to an existing file
+
+        Returns
+        -------
+        bool
+            True if given file is valid cwa file
+            False otherwise
+
+        Raises
+        ------
+        TypeError
+            if passed parameters are of incompatible type
+        """
+
+        if not isinstance(filename, str):
+            raise TypeError("filename must be a string")
+        ext = os.path.splitext(filename)
+        if ext[1] != ".cwa":
+            return False
+
+        f = open(filename, "rb")
+        tag = f.read(2).decode("ASCII")
+        if tag == "MD":
+            f.close()
+            return True
+        f.close()
+        return False
+
+    def __init__(self, filename, ):
+        if not isinstance(filename, str):
+            raise TypeError("filename must be a string")
+        self.__stream = open(filename, "rb")
+        self.__endianess = "<"
+        self.__meta = dict()
+        self.__device = 0
+        self.__name = 0
+        self.__uuid = 0
+        self.__startTime = datetime.min
+        self.__duration = timedelta()
+        self.__period = 0
+        self.__Accel = pandas.DataFrame(columns=["x","y", "z","norm"])
+        self.__Gyro = pandas.DataFrame(columns=["x","y", "z","norm"])
+        self.__Mag = pandas.DataFrame(columns=["x","y", "z","norm"])
+        self.__Ascale = 0
+        self.__Gscale = 0
+        self.__Mscale = 0
+        self.__Others = None
+
+        self.__blocks = 0
+        self.__stream.seek(0,2)
```

Comment:
@nbeliy : add a bit of comments when you do your magic. Here, I had to check the doc to understand you searched for the end of the file.

Reply:
It's a standard way to determine file size, I seen it as self-explanotary